### ISSUE TITLE
[v3] Implement new Concept Exercise: method-overloading

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#846]
This issue describes how to implement the `method-overloading` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Method Overloading in F#.

## Learning objectives

- Know what method overloading is.
- Know how to define overloaded methods.
- Know the limitations of method overloading.
- Know how to define optional parameters.
- Know how to pass named arguments.

## Out of scope

- Overload resolution.

## Concepts

This Concepts Exercise's Concepts are:

- `method-overloading`: know what method overloading is; know how to define overloaded methods; know the limitations of method overloading.
- `optional-parameters`: know how to define optional parameters.
- `named-arguments`: know how to use named arguments.

## Prerequisites

This Concept Exercise's prerequisites Concepts are:

- `classes`: know how to define methods on classes.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Overloaded methods][overloaded-methods]: what overloaded methods are and their restrictions.
- [Optional parameters][optional-parameters]: how to define optional parameters.
- [Named arguments][named-arguments]: how to pass arguments by name.

### After

- [Overloaded methods][overloaded-methods]: what overloaded methods are and their restrictions.
- [Optional parameters][optional-parameters]: how to define optional parameters.
- [Named arguments][named-arguments]: how to pass arguments by name.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[overloaded-methods]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/members/methods#overloaded-methods
[optional-parameters]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/parameters-and-arguments#optional-parameters
[named-arguments]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/parameters-and-arguments#named-arguments
